src/drivers/battery/bq40z50/bq40z50.cpp

```diff
@@ -785,7 +785,7 @@ int BQ40Z50::lifetime_read_block_one()
 {
 
 	uint8_t lifetime_block_one_size = 19;
-	uint16_t lifetime_block_one[lifetime_block_one_size] = {0};
```

Comment:
The problem is the 0, not the initializer. A single 0 would be for the first field only, not the entire array.

``` C
uint16_t lifetime_block_one[lifetime_block_one_size]{};
```